src/blueapi/worker/reworker.py

```diff
@@ -89,12 +89,24 @@ def __init__(
         self._stopping = Event()
         self._stopped = Event()
 
-    def clear_task(self, task_id: str) -> bool:
-        if task_id in self._pending_tasks:
-            del self._pending_tasks[task_id]
-            return True
-        else:
-            return False
+    def clear_task(self, task_id: str) -> str:
+        task = self._pending_tasks.pop(task_id)
+        return task.task_id
+
+    def cancel_active_task(
+        self,
+        cancel_type: WorkerState,
+        reason: Optional[str] = None,
+    ) -> str:
```

Comment:
I'm not in love with being able to pass a state that has nothing to do with stopping into the cancel method. Could have
```suggestion
    def cancel_active_task(
        self,
        failure: bool = False,
        reason: Optional[str] = None,
    ) -> str:
```

Reply:
mypy wasn't a fan of Literal[WorkerState.STOPPING, WorkerState.ABORTING], even after if new_state in {a, s}:...
This doesn't seem too bad. failure == new_state is WorkerState.ABORTING to have default = stop I guess. Will do now.